src/v/transform/transform_processor.cc

```diff
@@ -132,19 +132,20 @@ processor::processor(
 
 ss::future<> processor::start() {
     _as = {};
-    try {
-        co_await _engine->start();
-        co_await _source->start();
-        co_await _offset_tracker->start();
-    } catch (const std::exception& ex) {
-        vlog(_logger.warn, "error starting processor engine: {}", ex);
-        _state_callback(_id, _ntp, state::errored);
-    }
```

Comment:
presumably this wasn't important or is now handled by the caller?

Reply:
Source and offset tracker starts should never fail.

The engine failing is now handled async via `handle_processor_task`.

Coming around from our discussion about error handling, I think I want to swap the transform subsystem over to `result` to better communicate what is allowed to fail.